### PR TITLE
Fixes #7922: [API] Extend node information by "Last seen"

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/7922

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala [hunk 1]
```diff
@@ -562,12 +567,12 @@ class NodeApiService6 (
                        Full(Map[NodeId, Seq[Software]]())
                      }
     } yield {
-      val nodeIds = nodeFilter.getOrElse(nodeInfos.keySet)
       for {
         nodeId    <- nodeIds
         nodeInfo  <- nodeInfos.get(nodeId)
       } yield {
-        serializeInventory(nodeInfo, state, inventories.get(nodeId), software.getOrElse(nodeId, Seq()), detailLevel, version)
+        val runDate = runs.get(nodeId).map( _.map(_.agentRunId.date)).flatten
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala]: flatmap instead of first map + flatten ?

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala [hunk 2]
```diff
@@ -159,61 +161,63 @@ object NodeDetailLevel {
...
+  private val nodeInfoFields: Map[String, INFO => JValue] = {
+
+    val id           : INFO => JValue = (info:INFO) => info._1.id.value
+    val hostname     : INFO => JValue = (info:INFO) => info._1.hostname
+    val state        : INFO => JValue = (info:INFO) => info._1.state.name
+    val description  : INFO => JValue = (info:INFO) => info._1.description
+    val policyServer : INFO => JValue = (info:INFO) => info._1.policyServerId.value
+    val ram          : INFO => JValue = (info:INFO) => info._1.ram.map(MemorySize.sizeMb)
+    val arch         : INFO => JValue = (info:INFO) => info._1.archDescription
+    val runDate      : INFO => JValue = (info:INFO) => info._2.map(d => JString(DateFormaterService.getFormatedDate(d))).getOrElse(JNothing)
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala]: So here you won't have "lastRunDate" field if you ask for it and there was no run present, maybe we should always have a value, saying 'never' ? what do you think of this ?

Comment (fanf) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala]: I wasn't sure. It seems easier/more idiomatic for user to test for "field is null" than for an arbitrary value. But really, I'm not sure of idiomatism for these thing

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala (hunk 1 above)
Comment (fanf) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala]: done

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala (hunk 2 above)
Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala]: "idiotmatism"  would be fine for JSON ;)

What do our team members thinks ? what do people thinks ? what does the internet say ? let's look !

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala [hunk 3]
```diff
@@ -159,61 +161,63 @@ object NodeDetailLevel {
...
+        ( "version"       -> info._1.osDetails.version.value ) ~
+        ( "fullName"      -> info._1.osDetails.fullName ) ~
+        ( "servicePack"   -> info._1.osDetails.servicePack ) ~
+        ( "kernelVersion" -> info._1.osDetails.kernelVersion.value )
     }
 
     val machine = {
-      ( inv : NodeInfo ) =>
-        val (machineType,provider) = inv.machine.map(_.machineType match {
+      ( info : INFO ) =>
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala]: Here we could have unapplied the info to a tuple (inv, optLastRunDate ) but no pb il like the _1 too and would be quite PITA for you to rewrite it !!
